src/main/scala/io/citrine/lolo/bags/BaggedModel.scala

```diff
@@ -0,0 +1,76 @@
+package io.citrine.lolo.bags
+
+import breeze.linalg.DenseMatrix
+import io.citrine.lolo.Model
+
+import scala.collection.parallel.immutable.ParSeq
+
+/** A model holding a parallel sequence of models and the sample counts used to train them. */
+trait BaggedModel[+T] extends Model[T] {
+
+  /** Models in the ensemble trained on subsets of the training data. */
+  def ensembleModels: ParSeq[Model[T]]
+
+  override def transform(inputs: Seq[Vector[Any]]): BaggedPrediction[T]
+
+  override def shapley(input: Vector[Any], omitFeatures: Set[Int] = Set()): Option[DenseMatrix[Double]] = {
+    val ensembleShapley = ensembleModels.map(model => model.shapley(input, omitFeatures))
+    if (ensembleShapley.head.isEmpty) {
+      None
+    } else {
+      assert(ensembleShapley.forall(x => x.isDefined))
+
+      def sumReducer(a: Option[DenseMatrix[Double]], b: Option[DenseMatrix[Double]]): Option[DenseMatrix[Double]] = {
+        Some(a.get + b.get)
+      }
+
+      val scale = 1.0 / ensembleShapley.length
+      Some(scale * ensembleShapley.reduce(sumReducer).get)
+    }
+  }
+}
+
+class BaggedRegressionModel(
```

Comment:
Can these be case classes?

Reply:
Yeah, it's pretty inconsistent atm. I'll make them all case class.